hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieFileReaderFactory.java

```diff
@@ -46,14 +46,21 @@ public class HoodieFileReaderFactory {
   public static HoodieFileReaderFactory getReaderFactory(HoodieRecord.HoodieRecordType recordType) {
     switch (recordType) {
       case AVRO:
-        return new HoodieAvroFileReaderFactory();
+
+        try {
+          Class<?> clazz =
+              ReflectionUtils.getClass("org.apache.hudi.io.storage.HoodieHadoopAvroFileReaderFactory");
```

Comment:
Looks like the `HoodieFileReaderFactory` and `HoodieFileWriterFactory` are already the central station for creating all readers and writers.  We can add configs to control what `HoodieFileReaderFactory` and `HoodieFileWriterFactory` to return, and the current `HoodieFileReaderFactory` is for Hadoop-based implementation.

Reply:
create followup pr to do this